weather_mv/loader_pipeline/sinks.py

```diff
@@ -94,7 +145,12 @@ def open_dataset(uri: str, open_dataset_kwargs: t.Optional[t.Dict] = None, disab
                 shutil.copyfileobj(source_file, dest_file, DEFAULT_READ_BUFFER_SIZE)
                 dest_file.flush()
                 dest_file.seek(0)
-                xr_dataset: xr.Dataset = __open_dataset_file(dest_file.name, open_dataset_kwargs)
+
+                _, uri_extension = os.path.splitext(uri)
+                xr_dataset: xr.Dataset = __open_dataset_file(dest_file.name, uri_extension, open_dataset_kwargs)
+
+                if uri_extension == '.tif':
+                    xr_dataset = _preprocess_tif(xr_dataset, dest_file.name, tif_metadata_for_datetime)
```

Comment:
Since you have to re-open the file in the _preprocess_tif method anyway, I recommend handling all the opening here, and moving this block into an if-else block.

Reply:
Do you agree? My main reason is that opening the file might be a bit more efficient if done once.

Reply:
We are opening the TIF file in _preprocess_tif method using `rasterio` for getting the metadata. Filed a bug #159 to explore ways to get metadata using `xarray's` open_dataset method, in that case we will open TIF file only once in __open_dataset_file method.

Let’s discuss more on this on call.

Reply:
SGTM.